library/src/scala/tasty/reflect/TreeAccumulator.scala

```diff
@@ -0,0 +1,111 @@
+package scala.tasty
+package reflect
+
+/** TASTy Reflect tree accumulator.
+ *
+ *  Usage:
+ *  ```
+ *  class MyTreeAccumulator[R <: scala.tasty.Reflection & Singleton](val reflect: R)
+ *      extends scala.tasty.reflect.TreeAccumulator[X] {
+ *    import reflect.{given, _}
+ *    def foldTree(x: X, tree: Tree)(given ctx: Context): X = ...
+ *  }
+ *  ```
+ */
+trait TreeAccumulator[X] {
```

Comment:
Can we keep the following in `Reflection`:

```Scala
class TreeMap extends scala.tasty.reflect.TreeMap {
      val reflect: this.type = this
}
```

This way, programmers can just write:

```Scala
new TreeAccumulator[X] { ... }
class MyTreeAccumulator extends TreeAccumulator[X] { ... }
```

Instead of 

```Scala
class MyTreeAccumulator[R <: scala.tasty.Reflection & Singleton](val reflect: R)
extends scala.tasty.reflect.TreeAccumulator[X] { ... }

new scala.tasty.reflect.TreeMap {
    val reflect: qctx.tasty.type = qctx.tasty
    ...
 } 
```